### ISSUE TITLE
Fulltext search index: slow query in resetSearchResults()

### ISSUE THREAD

**Issue by arhu32** [magento/magento2#566]
In Magento/CatalogSearch/Model/Resource/__Fulltext.php__, function __resetSearchResults()__ there are some queries that set the whole column "is_processed" in the table "catalogsearch_query" to 0, e.g.:

```php
$adapter->update($this->getTable('catalogsearch_query'),
    array('is_processed' => 0));
```

This query is very slow because it possibly needs to update lots of rows every time resetSearchResults() is called. On our server (mage181, multi-store setup, no SSD) with ~40.000 rows in the table, the query took 0.5 seconds and was called regularly during product updates etc.

Fix:

1. Only update rows that are not already 0:

    ```php
    $adapter->update($this->getTable('catalogsearch/search_query'),
        array('is_processed' => 0), array('is_processed != 0'));
    ```

2. Add an index to the "is_processed" column in the "catalogsearch_query" table.

This will speed up the query considerably. AFAIK this issue affects all Magento versions, including 1.8.1.0.

**Comment by vpelipenko** [comment]
Internal ticket: MAGETWO-24366

**Comment by kokoc** [comment]
@arhu32 The fix is now available in 0.74.0-beta10. Thank you very much for your contribution and continued support to Magento 2! Could you please verify the issue?
